controllers/mlmd.go

```diff
@@ -33,16 +33,37 @@ func (r *DSPAReconciler) ReconcileMLMD(dsp *dspav1alpha1.DataSciencePipelinesApp
 
 	log.Info("Applying ML-Metadata (MLMD) Resources")
 
-	err := r.ApplyDir(dsp, params, mlmdTemplatesDir)
-	if err != nil {
-		return err
-	}
-
 	if params.UsingV1Pipelines(dsp) {
-		err = r.ApplyDir(dsp, params, mlmdTemplatesDir+"/v1")
+		if dsp.Spec.MLMD != nil {
+			if dsp.Spec.MLMD.Envoy == nil || dsp.Spec.MLMD.Envoy.DeployRoute {
+				err := r.ApplyDir(dsp, params, mlmdTemplatesDir)
+				if err != nil {
+					return err
+				}
+			} else {
+				err := r.ApplyDirExcept(dsp, params, mlmdTemplatesDir, []string{mlmdTemplatesDir + "/metadata-envoy.route.yaml.tmpl"})
```

Comment:
the logic here checks out, but for consistency we handled this in APIServer by simply putting the route YAML in its own directory and applying it if the `APIServer.EnableRoute` bool was true.  

I worry that this is more complicated than it needs to be, as there are now 'magic templates' that are conditionally applied vs other templates in the same directory, with no way to discern this w/o knowing the code's interal control flow.

APIServer control flow for comparison: https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/controllers/apiserver.go#L56-L60

Reply:
Yeah. Way better.
Done.